src/jit/lsra.cpp

```diff
@@ -3968,30 +3968,17 @@ void LinearScan::buildRefPositionsForNode(GenTree*                  tree,
 
             JITDUMP("t%u ", locInfo.loc);
 
-            // for interstitial tree temps, a use is always last and end;
-            // this is  set by default in newRefPosition
-            GenTree* useNode = locInfo.treeNode;
+            // for interstitial tree temps, a use is always last and end; this is set by default in newRefPosition
+            GenTree* const useNode = locInfo.treeNode;
             assert(useNode != nullptr);
-            var_types type        = useNode->TypeGet();
-            regMaskTP candidates  = getUseCandidates(useNode);
-            Interval* i           = locInfo.interval;
-            unsigned  multiRegIdx = locInfo.multiRegIdx;
 
+            Interval* const i = locInfo.interval;
             if (useNode->gtLsraInfo.isTgtPref)
             {
                 prefSrcInterval = i;
             }
 
-            const bool regOptionalAtUse = useNode->IsRegOptional();
-            const bool delayRegFree     = (hasDelayFreeSrc && useNode->gtLsraInfo.isDelayFree);
-
-            assert(isCandidateLocalRef(useNode) == i->isLocalVar);
```

Comment:
It was: I didn't feel that it added much value as this sort of condition should be checked elsewhere. I can put it back if you like.